include/inviwo/core/util/foreach.h

```diff
@@ -51,6 +51,19 @@ void foreach_helper(std::true_type, IT a, IT b, Callback callback, size_t start
     std::for_each(a, b, [&](auto v) { callback(v, start++); });
 }
 
+template <typename Callback, typename IT>
+auto foreach_helper_pool(std::true_type, IT a, IT b, Callback callback, size_t start = 0) {
+    return dispatchPool([=]() {
+        auto s = start;
```

Comment:
If a change to this I get `error: increment of read-only location ‘id’`
```c++
return dispatchPool([id=start,callback,a,b]() {
    std::for_each(a, b, [&](auto v) { callback(v, id++); });
});
```
(https://stackoverflow.com/a/27715105/600633)

Reply:
there is nothing wrong about making the lambda mutable, what is the problem? 
Think mating wanted to move the callback into the lambda to avoid the copy...
```
return dispatchPool([id=start, c = std::move(callback), a, b]() mutable {
    std::for_each(a, b, [&](auto v) { c(v, id++); });
});
```

Reply:
I was actually thinking of only capturing `s = start`